slack/backend/plugin/src/test/kotlin/com/ritense/slack/plugin/SlackPluginIT.kt

```diff
@@ -0,0 +1,227 @@
+
+        // then
+        val requestBody = getRequest(HttpMethod.POST, "/api/files.upload").body.readUtf8()
+        assertThat(requestBody).contains("AAAAA1111")
+        assertThat(requestBody).contains("Hello world!")
+        assertThat(requestBody).contains("tree.png")
+        assertThat(requestBody).contains("-tree-png-bytes-")
+    }
+
+    fun startMockServer() {
+        executedRequests = mutableListOf()
+        val dispatcher: Dispatcher = object : Dispatcher() {
+            @Throws(InterruptedException::class)
+            override fun dispatch(request: RecordedRequest): MockResponse {
+                executedRequests.add(request)
+                val response = when (request.method + " " + request.path?.substringBefore('?')) {
+                    "POST /api/chat.postMessage" -> mockResponseFromFile("/data/chat-post-message-response.json")
+                    "POST /api/files.upload" -> mockResponseFromFile("/data/files-upload-response.json")
+                    else -> MockResponse().setResponseCode(404)
+                }
+                return response
+            }
+        }
+        server = MockWebServer()
+        server.dispatcher = dispatcher
+        server.start()
+    }
+
+    fun findRequest(method: HttpMethod, path: String): RecordedRequest? {
+        return executedRequests
+            .filter { method.matches(it.method!!) }
+            .firstOrNull { it.path?.substringBefore('?').equals(path) }
+    }
+
+    fun getRequest(method: HttpMethod, path: String): RecordedRequest {
+        return findRequest(method, path) ?: fail("Request with method $method and path $path was not sent")
+    }
+
+    private fun createSlackPluginConfiguration(): PluginConfiguration {
+        val slackConfigurationProperties = """
+            {
+                "url": "${server.url("/")}",
+                "token": "test-token"
+            }"""
+
+        return pluginService.createPluginConfiguration(
+            "Slack plugin configuration",
+            Mapper.INSTANCE.get().readTree(slackConfigurationProperties) as ObjectNode,
+            "slack"
+        )
+    }
+
+    private fun createProcessLink(actionDefinitionKey: String, actionProperties: String) {
+        val processDefinition = repositoryService.createProcessDefinitionQuery()
+            .processDefinitionKey(PROCESS_DEFINITION_KEY)
+            .latestVersion()
+            .singleResult()
+
+        pluginService.createProcessLink(
+            PluginProcessLinkCreateDto(
+                processDefinition.id,
+                "ServiceTask",
+                configuration.id.id,
+                actionDefinitionKey,
+                Mapper.INSTANCE.get().readTree(actionProperties) as ObjectNode,
+            )
+        )
+    }
+
+    private fun createDocumentAndStartProcess(processVars: Map<String, Any> = emptyMap()) {
+        val documentContent = """
+            {
+                "lastname": "Klaveren"
```

Comment:
I would use a more generic lastname like "Doe" (as in John Doe)

Reply:
Changed